File: jmolecules-axon/02-onion-cqrs-es-bank/src/main/java/org/jmolecules/example/axonframework/application/ApplicationConfiguration.java
package org.jmolecules.example.axonframework.application;

import org.jmolecules.example.axonframework.domain.command.model.MoneyTransferIdGenerator;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class ApplicationConfiguration {

  @Bean
  public MoneyTransferIdGenerator randomGenerator() {
    return MoneyTransferIdGenerator.RANDOM;
  }
}

File: jmolecules-axon/03-onion-axon-cqrs-es-bank/src/main/kotlin/org/jmolecules/example/axonframework/core/application/ApplicationConfiguration.kt
package org.jmolecules.example.axonframework.core.application

import org.jmolecules.architecture.onion.classical.ApplicationServiceRing
import org.springframework.context.annotation.ComponentScan
import org.springframework.context.annotation.Configuration

@Configuration
@ComponentScan
@ApplicationServiceRing
class ApplicationConfiguration
